### ISSUE TITLE
Renaming "admin" account slows down guest access

### ISSUE THREAD

**Issue by onovy** [janeczku/calibre-web#2655]
**Describe the bug/problem**
This will probably be most funny report i had ever reported, but I tested it many times in venv and docker env. I can reproduce it with clean environment.

There is one default user "admin". When username is changed to anything else, guest access is slowed down 10 times.

Measured latencies for "/cover/id/md" downloading.

Admin username is "admin":
Logged in: ~100 ms
Logged out: ~100 ms

Admin username is anything else:
Logged in: ~100 ms
Logged out: **~1000 ms**

**To Reproduce**
1. Allow guest access
2. Change username of default "admin" user to anything else
3. Logout
4. Check latency, example URL: https://example.com/cover/89/md?c=1673528171

**Logfile**
Nothing relevant in logfile even with "debug".

**Expected behavior**
Don't slow down application.

**Environment:**
 - OS: Linux/Ubuntu or Docker
 - Python version: 3.8.10
 - Calibre-Web version: 0.6.19
 - Docker container: reproducable inside and outside of Docker
 - Special Hardware: nope
 - Browser: All browsers

Thanks.

**Comment by OzzieIsaacs** [comment]
I doublechecked it with a quite big library. Independent of the name of the admin the page loads with the same speed.
I also created a new admin user and deleted the inital on, no change

**Comment by onovy** [comment]
I guess figured it out. I'm running calibre-web behind nginx proxy, which does HTTP Basic Auth. For security reason I want to authentication on web server first but don't want to have same users inside Calibre-web. I'm using Calibre-web as view-only, so guest view is fine for me.

When I logged out from Calibre-web, Calibre-Web see Authorization header (proxy_passed from nginx) and tries to authenticate (https://github.com/janeczku/calibre-web/blob/master/cps/usermanagement.py#L61). When username in HTTP Basic Auth is different than in application DB, application don't find entry in DB and everything is fine. But when user is found, it will check password which is PBKDF2 stretched, which takes time.

I removed Authorization header in nginx from proxy_passed request and everything is fine now.

Sorry for really strange bug report.
